quickjs.c

```diff
@@ -57422,6 +57423,144 @@ static void insert_weakref_record(JSValueConst target,
     *pwr = wr;
 }
 
+/* urlsafe_base64 atob/btoa */
+
+// Base64 encoding table
+static const char base64_table[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_";
+
+// btoa: binary to ASCII (base64 encode)
+char* btoa(JSContext *ctx, const char* bin, size_t len) {
+    if (bin == NULL || len == 0) return NULL;
+    // Calculate output length (including padding)
+    size_t out_len = 4 * ((len + 2) / 3) + 1;
+    // Allocate memory for output string (plus null terminator)
+    char* out = (char*)js_mallocz(ctx, out_len);
```

Comment:
Unnecessary cast but more substantially, allocating should preferably be done by the caller, because then the caller can use e.g. a stack-allocated buffer for small inputs.